### ISSUE TITLE
NonInteractiveCliktConsole only reads in the first line from an input file

### ISSUE THREAD

**Issue by JLLeitschuh** [ajalt/clikt#32]
Currently the `NonInteractiveCliktConsole` only reads in the first line of any input file passed as the input stream.

This can be fixed by caching the buffered reader like I have below.

```kotlin
class NonInteractiveCliktConsole : CliktConsole {
    private val bufferedReader by lazy {
        System.`in`.bufferedReader()
    }

    override fun promptForLine(prompt: String, hideInput: Boolean) = try {
        print(prompt, false)
        bufferedReader.readLine()
    } catch (err: IOException) {
        null
    }

    override fun print(text: String, error: Boolean) {
        if (error) {
            System.err
        } else {
            System.out
        }.print(text)
    }

    override val lineSeparator: String get() = System.lineSeparator()
}
```

I'm using version `1.5.0`.

**Comment by ajalt** [comment]
Thanks!

**Comment by JLLeitschuh** [comment]
I actually think that `NonInteractiveCliktConsole` needs to be a singleton, or at least the `bufferedReader` field needs to be a singleton. Otherwise the first `NonInteractiveCliktConsole` that is created will consume the entire buffer. Thoughts?

**Comment by ajalt** [comment]
Yes, I think you're right. I could add a reader to the command context that could be shared between parameters.

**Comment by grote** [comment]
I am still seeing this issue on Linux with version 1.7.0. Neither a multiline bash pipe or sending data with Python works. I am always getting this exception:

```ruby
Exception in thread "main" java.lang.RuntimeException: EOF
        at com.github.ajalt.clikt.output.NonInteractiveCliktConsole.promptForLine(CliktConsole.kt:63)
        at com.github.ajalt.clikt.output.TermUi.prompt(TermUi.kt:94)
        at com.github.ajalt.clikt.output.TermUi.prompt$default(TermUi.kt:86)
```

More information: https://code.briarproject.org/briar/briar/issues/1512

Any ideas/hints are welcome!

**Comment by grote** [comment]
Here's a minimal sample to try:

```kotlin

import com.github.ajalt.clikt.core.CliktCommand
import com.github.ajalt.clikt.output.TermUi.prompt

class Cli : CliktCommand() {

    override fun run() {
        prompt("test1")
        prompt("test2")
    }

}

fun main(args: Array<String>) = Cli().main(args)
```

How to pipe two lines here without getting a EOF exception?

**Comment by ajalt** [comment]
I'll take a look. Given that our travis builds run on linux, I'm surprised that the behavior is different. I'd love suggestions on how to improve the test suite so that it can reproduce this problem.

**Comment by grote** [comment]
Yeah I just ran the tests myself in the hope of finding the issue, but they passed for me, even when using `stdin.provideText("test\ntest\n")` instead of the line joining.

When modifying one of the samples to use the code above, I could reproduce the problem though by running:

```bash
echo "test
test
" | ./runsample prompt
```
or
```bash
echo "test\ntest\n" | ./runsample prompt
```
Before thinking about how to add a test here, maybe let's see if you can reproduce the problem.

**Comment by ajalt** [comment]
Yes, I'm able to reproduce it from the command line.

**Comment by ajalt** [comment]
So I'm finding that redirecting stdin at all fails to work. For example:
```kotlin
fun main() {
    println(System.`in`.bufferedReader().readLines())
}
```

prints `[]` when called with data piped to stdin, regardless of how many lines you give it. I'll have to do some mroe investigating.

**Comment by ajalt** [comment]
Thanks for the reports! This is fixed in 4425fe2, which is part of the 2.0.0 release.
